### ISSUE TITLE
tf.keras compatibility: convert multi-label-classification to Keras Core

### ISSUE THREAD

**Issue by Akorex** [keras-team/keras#18438]
for tf.keras backwards compatibility

In my attempt to port [Large-scale multi-label classification](https://keras.io/examples/nlp/multi_label_classification) to `keras_core` I hit an issue with the Inference Section. The example from keras.io created a new Inference model after training to include the TextVectorization preprocessing layer so inference can be done by passing strings directly to the model. 

```model_for_inference = keras.Sequential([text_vectorizer, shallow_mlp_model])```

This didn't work for keras_core imports and the attempt to port the example didn't include the particular step from the example. 

PR: keras-team/keras-core#538 

Colab Notebook: https://colab.research.google.com/drive/1kR7W4BJhtDlA5Jx_e4EwUM2FnzKP-D58#scrollTo=xmf6rTLL0G9H

**Comment by sonali-kumari1** [comment]
Hi @Akorex,

Are you still able to reproduce this issue ? If yes, could you please provide accessible link for colab notebook as i am getting file does not exist error. Thanks!
